src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala

```diff
@@ -179,7 +179,17 @@ abstract class ClassfileWriters {
           case _: FileAlreadyExistsException => FileChannel.open(path, fallbackOpenOptions)
         }
 
-        os.write(ByteBuffer.wrap(bytes), 0L)
+        try {
+          os.write(ByteBuffer.wrap(bytes), 0L)
+        } catch {
+          case ex: ClosedByInterruptException =>
+            try {
+              Files.deleteIfExists(path) // don't leave a empty of half-written classfile around after an interrupt
```

Comment:
I noticed errors on subsequent compiles after compilation like `class file .../package.class is broken / ArrayIndexOutOutBoundsException(0)`. My guess was I was seeing a file that had been `open`-ed but not written to.

Reply:
Maybe we need to look more closely at what SBT/Zinc does with its management of classfiles on interruption.

Reply:
I think the transactional classfile manager will detect that zinc failed, and restore the class files that were available before compiling. So within Zinc, this should not be a problem.